dymos/phase/phase.py

```diff
@@ -1211,7 +1231,19 @@ def add_path_constraint(self, name, constraint_name=None, units=None, shape=None
             If True, treat indices as flattened C-ordered indices of elements to constrain at each given point in time.
             Otherwise, indices should be a tuple or list giving the elements to constrain at each point in time.
         """
-        if constraint_name is None:
+        expr_operators = ['(', '+', '-', '/', '*', '&', '%', '@']
+        if '=' in name:
+            is_expr = True
+        elif '=' not in name and any(opr in name for opr in expr_operators):
+            raise ValueError(f'The expression provided {name} has invalid format. '
+                             'Expression may be a single variable or an equation'
+                             'of the form "constraint_name = func(vars)"')
+        else:
```

Comment:
I question the need for all of this expression checking (here) and the name checking (in introspection). I think OpenMDAO might do all this checking in the execcomp, so these might be redundant. The only reason you might do this here is to deliver a better error message to the user.